impl_core/src/main/java/io/opencensus/implcore/stats/RecordUtils.java

```diff
@@ -83,15 +100,29 @@
     for (int i = 0; i < columns.size(); ++i) {
       TagKey tagKey = columns.get(i);
       if (!tags.containsKey(tagKey)) {
-        // replace not found key values by null.
-        tagValues.add(UNKNOWN_TAG_VALUE);
+        @javax.annotation.Nullable TagValue tagValue = UNKNOWN_TAG_VALUE;
+        if (RPC_TAG_MAPPINGS.containsKey(tagKey)) {
+          tagValue = getTagValueForDeprecatedRpcTag(tags, tagKey);
+        }
+        tagValues.add(tagValue);
       } else {
         tagValues.add(tags.get(tagKey).getTagValue());
       }
     }
     return tagValues;
   }
 
+  // TODO(songy23): remove the mapping once we completely remove the deprecated RPC constants.
+  private static TagValue getTagValueForDeprecatedRpcTag(
+      Map<? extends TagKey, TagValueWithMetadata> tags, TagKey oldKey) {
+    for (TagKey newKey : RPC_TAG_MAPPINGS.get(oldKey)) {
```

Comment:
@carl-mastrangelo, I think the for-each construct will create an Iterator, which you may not want to see. Maybe `RPC_TAG_MAPPINGS` should be defined as a `Map<TagKey, TagKey[]>` because for-each on an array presumably doesn't create an Iterator?